sites/geohub/src/routes/settings/+page.server.ts

```diff
@@ -7,9 +7,17 @@ export const actions = {
       return fail(403, { message: 'No permission' })
     }
     const data = await request.formData()
-    const settings: { [key: string]: string } = {}
-    for (const [key, value] of data.entries()) {
-      settings[key] = value
+    const settings = {
+      SidebarPosition: data.get('SidebarPosition'),
+      SearchLimit: parseInt(data.get('SearchLimit')),
+      DatasetSearchLimit: parseInt(data.get('DatasetSearchLimit')),
+      DatasetSearchQueryOperator: data.get('DatasetSearchQueryOperator'),
+      DatasetSortingColumn: data.get('DatasetSortingColumn'),
+      DataPageSortingColumn: data.get('DataPageSortingColumn'),
+      MapSortingColumns: data.get('MapSortingColumns'),
+      TagSearchOperator: data.get('TagSearchOperator'),
+      NumberOfClasses: parseInt(data.get('NumberOfClasses')),
+      LineWidth: parseFloat(data.get('LineWidth')),
```

Comment:
Can you do it in other way rather than specifying all properties' name to convert data type?

maybe you can check property's data type from DefaultUserConfig's value dynamically